### ISSUE TITLE
Shaka packager wrapper for IOS

### ISSUE THREAD

**Issue by denandreychuk** [shaka-project/shaka-packager#820]
I want to use raw_key_encryption from Shaka Packager directly in my IOS App to generate DRM videos offline. Is there a way how I can do it? Is there any solution for this?

**Comment by kqyang** [comment]
Do you mean running Shaka Packager library or binary in iOS? May I know why you want to do that?

**Comment by denandreychuk** [comment]
@kqyang thank for your reply! Correct. I want to run Shaka Packager in my IOS app to create DRM video from user content. I need only one feature. It's --enable_raw_key_encryption. No other features required.

`packager \
  in=1.mp4,stream=video,output=2.mp4,drm_label=SD \
  --enable_raw_key_encryption \
  --keys label=SD:key_id=abba271e8bcf552bbd2e86a434a9a5d9:key=69eaa802a6763af979e8d1940fb88392
`

So basically, I want to grab for example user picture, create video from this picture (this I can achieve with Apple API) and then add simple raw key DRM protection to this content (this is why I want to use Shaka).

**Comment by denandreychuk** [comment]
This allows me to protect content from screenshots (Apple hide DRM protected content on screenshots. This is for example how Netflix do the same in their own app). I can achieve this using FairPlay and custom server setup, but it a huge overhead. So I was wondering is there a way to wrap only this feature from Shaka, maybe somehow using Objective-C++ or etc.

**Comment by kqyang** [comment]
I see. It might be possible to cross compile Shaka Packager on iOS, but you'll probably have to explore it yourself. See the instructions here for cross compiling for ARM: https://google.github.io/shaka-packager/html/build_instructions.html#cross-compiling-for-arm-on-ubuntu-host, which may be extended to cross compile Shaka Packager on iOS.

Let us know if you have any questions.
